### PR TITLE
[REG-1332] feat: move uns tlds minting config outside of minting manager

### PR DESCRIPTION
**Pull request by roman-gnativ**
## PR Checklist

**Linear ticket:** 
https://linear.app/unstoppable-domains/issue/REG-1332/introduce-tld-configs-in-uns-repo

**Background:** 
To deploy our registry contracts to the base chain, we have made UNS contracts unified across chains, which, in the case of the base chain, means removing the TLD config outside the minting manager contract. 

**What was done:**
- Tld config removed from the Minting manager smart contract
- Tld config introduced in the uns repo
- Added a deployment task that mints all TLDs from the TLD config

### 1. Contracts versioning
- [ ] Make sure that the `patch` version of the contracts is increased if changes have been made to the `UNSRegistry`, `MintingManager`, `ProxyReader`, `ENSCustody` contracts.
- [ ] Make sure that the `minor` version of the contracts is increased if breaking changes have been made to the `UNSRegistry`, `MintingManager`, `ProxyReader`, `ENSCustody` contracts. It includes changes of interfaces.
### 2. Contracts licensing
- [ ] Make sure that no **SPDX-License-Identifier** defined in contracts.
- [ ] Make sure that the **header** is added to the new contract files.
  ```
  // @author Unstoppable Domains, Inc.
  // @date {Month} {Day}(ordinal), {Year}
  ```
### 3. Coverage
- [ ] Make sure that the coverage of contracts has not decreased and strive **100%**
### 4. Configs versioning
- [ ] Make sure that the version of `uns-config.json` is increased if changes have been made to the config.
- [ ] Make sure that the version of `ens-config.json` is increased if changes have been made to the config.
- [ ] Make sure that the version of `resolver-keys.json` is increased if changes have been made to the config.
- [ ] Make sure that the version of `ens-resolver-keys.json` is increased if changes have been made to the config.
### 5. Package versioning
- [ ] Make sure that the `patch` version of package is increased if valuable changes have been made to the package. It includes contracts update, configs update, etc.
- [ ] Make sure that the `major.minor` version of package is synced with version of `UNSRegistry` contract.
- [ ] Make sure that the `CHANGELOG` is updated with short description for the new version.
### 6. Code review
- [ ] `resolver-keys.json` code review is required from **DevTools** team
- [ ] `ens-resolver-keys.json` code review is required from **DevTools** team
- [ ] For all other changes code review is required from **Registry** team

### CODE REVIEW

**Review by DeRain — APPROVED** [review]

File: test/helpers/constants.ts [hunk 1]
```diff
@@ -9,46 +11,315 @@ export const DAY = 24 * 60 * 60;
...
-  METROPOLIS: BigInt('0x46a29e7d78413806ab5b6e30bc360bbff329e7bec2fc3ddf6f62d8c6c17c45b6'),
-  WRKX: BigInt('0xacf08822cf871e802cb23fdb2224cb924878985b866b92e55aaae8c0a8a898eb'),
-  SECRET: BigInt('0x335f1d8a43ebd6d9277127e859f502a14bbe3ccbe2d3f89be38fd0086da4c396'),
-  RAIIN: BigInt('0x2418c6dc1c5996890c54f1b197ce3d5f255b857445bce309215d0751b56fcd65'),
-  STEPN: BigInt('0x05cb1b6d0e0a38c2fdeacf5d8383eea8f73d8fdeca1e4cc99a0e202f68d5b7df'),
-
-  // Expirable
-  COM: BigInt('0xac2c11ea5d4a4826f418d3befbf0537de7f13572d2a433edfe4a7314ea5dc896'),
-  CA: BigInt('0xc9dca3620bc4f436a4161d86f335af2a8f7eb7bae05865a71e18fc617ef98c4e'),
+export enum SupportedNetworks {
```
Comment (nickshatilo) [inline test/helpers/constants.ts:14]: maybe simply `Networks`?

File: test/helpers/constants.ts [hunk 2]
```diff
@@ -9,46 +11,315 @@ export const DAY = 24 * 60 * 60;
...
+  nameServices: NameService[];
+  chainIds: SupportedNetworks[];
+  hash: bigint;
+  expirable?: boolean;
+}
+
+export const TLD: Record<string, TLDConfig> = {
+  crypto: {
+    nameServices: [NameService.UNS],
+    chainIds: [
```
Comment (nickshatilo) [inline test/helpers/constants.ts:36]: can we do rename to `networks`? since it's actually consists of those

File: src/helpers.ts [hunk 3]
```diff
@@ -55,3 +63,36 @@ export const upgradeProxy = async <T extends BaseContract>(
...
+};
+
+export function getExpirableTlds (): string[] {
+  return Object.entries(TLD)
+    .filter(([, tldConfig]) => tldConfig.expirable)
+    .filter(([, tldConfig]) => applyfilterTldsByChainId(tldConfig))
+    .map(([tld]) => tld);
+}
+
+function applyfilterTldsByChainId (tldConfig: TLDConfig): boolean {
```
Comment (nickshatilo) [inline src/helpers.ts:86]: `applyFilter`?

File: src/helpers.ts [hunk 4]
```diff
@@ -3,9 +3,17 @@ import {
...
-import type { BaseContract, ContractFactory } from 'ethers';
+import { BaseContract, ContractFactory, ethers } from 'ethers';
 import { network, upgrades } from 'hardhat';
-import { DependenciesMap, ContractName, NsNetworkConfig } from './types';
+import { MintingManager } from '../types';
+import { NetworkChainIds, TLD, TLDConfig } from '../test/helpers/constants';
+import { ContractName, DependenciesMap, NsNetworkConfig } from './types';
+import { NameService } from './config';
+
+export const isSandbox = network.config.chainId === 1337 || network.config.chainId === 31337;
```
Comment (nickshatilo) [inline src/helpers.ts:13]: shall we maybe just have two arrays of `SANDBOX_NETWORK_IDS` and `TESTNET_NETWORK_IDS` and just check in those?

**Comment by nickshatilo** [comment]
overall LGTM @roman-gnativ, just a few naming / structure suggestions

**Review by nickshatilo — APPROVED** [review]

File: ens-config.json [hunk 5]
```diff
@@ -73,45 +73,45 @@
         "1337": {
```
Comment (nickshatilo) [inline ens-config.json:73]: please increase version of ens-config

File: uns-config.json [hunk 6]
```diff
@@ -359,9 +359,9 @@
                     "deploymentBlock": "0x0f"
                 },
                 "DotCoinBurner": {
-                    "address": "0x3F1194B9019002863Ad0DBBE295294547c0d36A9",
+                    "address": "0x287DeE389Caaa49edc112888f2BcFD927b472e25",
                     "legacyAddresses": [],
-                    "deploymentBlock": "0x32"
```
Comment (nickshatilo) [inline uns-config.json:364]: same here

**Pull request merged**